cmake/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+# The following lines of boilerplate have to be in your project's
+# CMakeLists in this exact order for cmake to work correctly
+cmake_minimum_required(VERSION 3.5)
+
+add_definitions(
+  # this option should be used, but OpenMRN does not compile
+  # unfortunately all the network functions are non-blocking/polling when
+  # using the arduino interface
+  #-D__FreeRTOS__    
```

Comment:
The ESP32 intentionally does not use ```__FreeRTOS__``` for OpenMRN. The majority of the OpenMRN code does internally use FreeRTOS functionality on the ESP32 but it is not controlled via this flag.

The network IO being non-blocking is intentional for all platforms that support it (including ESP32). On these platforms the select() model is used, on the ESP32 there is an implementation of this from ESP-IDF that has been augmented for cross-thread notification as used by OpenMRN.

Reply:
I noticed that the socket accept call within `lib/OpenMRNLite/src/utils/socket_listener.cpp` `void SocketListener::AcceptThreadBody()` was timing out every 100ms and saw that this wouldn't happen with the `__FREE_RTOS__` define set (openmrn_features.h is then not setting `OPENMRN_HAVE_BSD_SOCKETS_RX_TIMEOUT`). I locally changed my code to exclude the setsockopt call. This yields noticeably faster (re-)connection with JMRI.

Reply:
That sounds like a bug, possibly in the detection of a closed connection. It should be using a 100uS timeout on RX and if the socket is still valid it should just go back into a blocked state waiting for data. Is JMRI disconnecting?

Reply:
This is happening in the accept call, so before anybody connects. It will go back into the accept call, just shortly coming out it. Not a big deal, even though I believe to see a difference in response.

Reply:
> This is happening in the accept call, so before anybody connects.

Hmm, this may have other unintended consequences, the RX timeout should be enforced on all socket connections inside OpenMRN and not just for the accept side of it. It sounds like there may be a small gap in the functionality which should be reviewed further.

One thing is certain though, this flag can not be passed in as it will enable other functionality which will not be guaranteed to work on the ESP32. It is intentionally not defined on this platform.

Reply:
thanks for the explanation - good to know. I removed the comment from the cmake file